### DISCUSSION TITLE
[MPCTools] How to use getCasadiDAE.m to add algebraic constraints?

### DISCUSSION THREAD

**Discussion by CanChen-Caesar** [casadi/casadi#3072] [Q&A] [answered]
Hello,

I am trying to add a constraint to the state [x1, x2] and the control input [u1, u2], like

x1+u1=c1,
x2+u2=c2

where c1 and c2 are fixed constants. I am wondering if any examples show how to use getCasadiDAE.m to make this constraint work.

Regards and thanks,
Caesar

**Comment by jgillis** [comment, marked as answer]
What are the differential and algebraic states here?

The description looks like both x1 and x2 are algebraic states and trivially, `x1=c1-u1`, `x2=c2-x2`?
Once you identified algebraic states `z` and algebraic constraints, you may follow the example in https://bitbucket.org/rawlings-group/octave-mpctools/src/4e9398f1c35a6e09d9bda065f4e2afb5724d3dea/unittests/testdae.m#lines-27 to create an extra argument for `getCasadiDAE.m`

**Reply by CanChen-Caesar** [reply]
Thank you. This is really helpful.
